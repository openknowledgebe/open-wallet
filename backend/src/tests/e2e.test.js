/* eslint-disable import/no-extraneous-dependencies */
// import our production apollo-server instance
const gql = require('graphql-tag');
const { server, db } = require('../');

const { startTestServer, toPromise, populate } = require('./utils');

const GET_ME = gql`
  query me {
    me {
      name
      id
    }
  }
`;

const LOGIN_ME_IN = gql`
  mutation login($email: String!, $password: String!) {
    login(email: $email, password: $password) {
      name
      id
    }
  }
`;

describe('Server - e2e', () => {
  let stop;
  let graphql;

  beforeAll(async () => {
    await db.connect();
    await populate();
  });

  afterAll(async () => {
    await db.disconnect();
  });

  beforeEach(async () => {
    const testServer = await startTestServer(server);
    // eslint-disable-next-line prefer-destructuring
    stop = testServer.stop;
    // eslint-disable-next-line prefer-destructuring
    graphql = testServer.graphql;
  });

  afterEach(async () => stop());

  describe('Me query', () => {
    it('fails when user not logged in', async () => {
      const res = await toPromise(
        graphql({
          query: GET_ME
        })
      );
      expect(res).toMatchSnapshot();
    });
  });

  describe('Login', () => {
    it('fails on incorrect email', async () => {
      const res = await toPromise(
        graphql({
          query: LOGIN_ME_IN,
          variables: { email: 'email@gmail.com', password: 'johnnnyyy' }
        })
      );
      expect(res).toMatchSnapshot();
    });
    it('fails on incorrect password', async () => {
      const res = await toPromise(
        graphql({
          query: LOGIN_ME_IN,
          variables: { email: 'Sylvie@hotmail.com', password: 'johnnnyyy' }
        })
      );
      expect(res).toMatchSnapshot();
    });

    it('succeeds', async () => {
      const res = await toPromise(
        graphql({
          query: LOGIN_ME_IN,
          variables: { email: 'Sylvie@hotmail.com', password: 'sylvie2324' }
        })
      );
      expect(res).toMatchSnapshot();
    });
  });

  // TODO test registration
});
