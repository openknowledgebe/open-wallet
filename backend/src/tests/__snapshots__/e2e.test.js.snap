// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Server - e2e Login fails on incorrect email 1`] = `
Object {
  "data": null,
  "errors": Array [
    Object {
      "extensions": Object {
        "code": "UNAUTHENTICATED",
      },
      "locations": Array [
        Object {
          "column": 3,
          "line": 2,
        },
      ],
      "message": "Incorrect email or password. Please try again.",
      "path": Array [
        "login",
      ],
    },
  ],
}
`;

exports[`Server - e2e Login fails on incorrect password 1`] = `
Object {
  "data": null,
  "errors": Array [
    Object {
      "extensions": Object {
        "code": "UNAUTHENTICATED",
      },
      "locations": Array [
        Object {
          "column": 3,
          "line": 2,
        },
      ],
      "message": "Incorrect email or password. Please try again.",
      "path": Array [
        "login",
      ],
    },
  ],
}
`;

exports[`Server - e2e Login succeeds 1`] = `
Object {
  "data": Object {
    "login": Object {
      "id": "5ca36073a89e1e5c7ad111af",
      "name": "Sylvie Delcourt",
    },
  },
}
`;

exports[`Server - e2e Me query fails when user not logged in 1`] = `
Object {
  "data": Object {
    "me": null,
  },
  "errors": Array [
    Object {
      "extensions": Object {
        "code": "UNAUTHENTICATED",
      },
      "locations": Array [
        Object {
          "column": 3,
          "line": 2,
        },
      ],
      "message": "You must be logged in.",
      "path": Array [
        "me",
      ],
    },
  ],
}
`;
